### PR TITLE
Development fixes and small release-assignment improvement

### PR DESCRIPTION
**Pull request by joeyAghion**
Please review by commit, since I had to modify a few things to get this work started. These steps included:
* Some developer environment fixes to be able to run locally.
* Refactored `DeployService` operations into instance methods for better encapsulation and memoization.
* If a release PR already exists, but has no assignee, check if there's a good candidate and assign them.

This^ last problem was my reason for coming here. Often, the initial diff in a release PR is contributed by bots, and human contributors' commits get added later. Until now, Horizon only chose an assignee when creating the release PR (so based only on those initial commits). Going forward, it will add an assignee to an existing release PR when human contributions get added.

### CODE REVIEW

File: spec/features/comparisons_spec.rb [hunk 1]
```diff
@@ -57,7 +57,7 @@
       allow_any_instance_of(Releasecop::Checker).to receive(:check).and_return(
         Releasecop::Result.new('shipping', [large_comparison])
       )
-      expect(DeployService).to receive(:start)
+      expect_any_instance_of(DeployService).to receive(:start)
```
Comment (starsirius) [inline spec/features/comparisons_spec.rb:60]: 🙈

File: spec/features/deploy_spec.rb [hunk 2]
```diff
@@ -68,4 +68,25 @@
...
   end
+
+  it 'adds assignees to existing deploy PRs when unassigned' do
+    expect_any_instance_of(Octokit::Client).to receive(:create_pull_request)
+      .with('artsy/candela', 'release', 'staging', anything, anything)
+      .and_raise(Octokit::UnprocessableEntity)
+    allow_any_instance_of(Octokit::Client).to receive(:pull_requests)
+      .with('artsy/candela', base: 'release', head: 'staging')
+      .and_return([double(number: 42, assignee: nil)])
+    allow_any_instance_of(Octokit::Client).to receive(:pull_request_commits)
```
Comment (starsirius) [inline spec/features/deploy_spec.rb:79]: #minor. Maybe we can always use `expect_...` to be stricter?

File: README.md [hunk 3]
```diff
@@ -18,12 +18,16 @@ Visual representations of release pipelines.
...
     hokusai dev start
-    # or
-    dotenv bundle exec rails db:prepare
-    dotenv bundle exec rails server
+
+Or on localhost:
+
+    bundle exec rails db:prepare
+    yarn install
+    bundle exec rails server
```
Comment (starsirius) [inline README.md:30]: Looks like local development doesn't need any special/non-default env variables?

Comment (joeyAghion) [reply README.md:30]: Correct that's the intention.

File: spec/features/deploy_spec.rb (hunk 2 above)
Comment (joeyAghion) [reply spec/features/deploy_spec.rb:79]: I didn't want to concern this spec too much with _how_ the assignee is chosen, only that there is one.

**Pull request merged**